ons_alpha/private_media/models.py

```diff
@@ -0,0 +1,217 @@
+import logging
+
+from collections.abc import Iterable
+
+from botocore.exceptions import ClientError
+from django.core.files.base import File
+from django.db import models
+from django.utils import timezone
+from wagtail.models import Collection, CollectionMember
+
+from ons_alpha.private_media.utils import (
+    get_private_root_collection_id,
+    get_private_root_collection_path,
+    get_public_root_collection_id,
+    get_public_root_collection_path,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class ProtectedCollection(models.Model):
+    collection = models.OneToOneField(Collection, on_delete=models.PROTECT)
+
+    def __str__(self):
+        return f"Exists to prevent accidental deletion of {self.collection!r}"
+
+
+class PrivateMediaMixin(models.Model):
+    is_private = models.BooleanField(default=False, editable=False)
+    last_public_collection = models.ForeignKey(
+        Collection, null=True, editable=False, on_delete=models.SET_NULL, related_name="+"
+    )
+    last_private_collection = models.ForeignKey(
+        Collection, null=True, editable=False, on_delete=models.SET_NULL, related_name="+"
+    )
```

Comment:
Issue: Currently, if a user moves an object into a collection below one of these (say "Private" to "Private / Page 1"), `last_private_collection` isn't updated. It will be if the permission changes, but otherwise it's not. I'm not sure how much of a concern that is, but it's be nice to handle it.

Reply:
Nice catch! Have updated `determine_privacy()` to set them regardless of a 'change in privacy' now